languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -96959,6 +96959,82 @@ The accident victim died from her injuries.
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes">\-</token>
+                </antipattern>
+                <antipattern>
+                    <token skip="-1"></token>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes">\-</token>
+                    <token>maybe</token>
+                    <token><match no="0"/></token>
+                </antipattern>
+                <antipattern>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes" skip="-1">\-</token>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes" skip="-1">\-</token>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes" min="0">\-</token>
+                </antipattern>
+                <antipattern>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes" skip="-1">\-</token>
+                    <token regexp="yes">.*[a-z].*</token>
+                    <token regexp="yes">\.</token>
+                    <token regexp="yes">\d+</token>
+                    <token regexp="yes">\.</token>
+                    <token regexp="yes" skip="-1">\d+</token>
+                    <token regexp="yes">\-</token>
+                    <token regexp="yes">\-</token>
+                </antipattern>
+                <pattern>
+                        <marker>
+                            <token regexp="yes" spacebefore="yes">\-</token>
+                            <token regexp="yes">\-</token>
+                            <token spacebefore="yes" skip="-1"></token>
+                            <token regexp="yes" spacebefore="yes">\-</token>
+                            <token regexp="yes">\-</token>
+                        </marker>
+                        <token spacebefore="yes" min="0"></token>
+                </pattern>
+                <message>Consider using an em dash (<suggestion>—</suggestion>) or an en dash with spaces (<suggestion> – </suggestion>) instead of '--'.</message>
+                <example correction="—| – ">This is <marker>-- as far as I can tell --</marker> a very interesting story.</example>
+                <example correction="—| – ">No one <marker>-- not even his closest friends --</marker> could cheer him up.</example>
+                <example correction="—| – ">This is <marker>-- alas --</marker> all I can do.</example>
+                <example>------ ----------- ----------- --------- --------- ---------------- </example>
+                <example>draw.io -- Window Properties Report -- 01 Windows Listed</example>
+                <example>Open -- System Preferences -- Notifications.kmmacros (6.4 KB)</example>
+                <example>Download: Global -- Bear -- New Note v1.0.kmmacros</example>
+                <example>https://' 'Large_Doc_Test -- Very Large Doc -- 500,000 Lines</example>
```

Comment:
Just out of curiosity: Why did you choose to exclude these contexts?

Reply:
I thought that these were cases where the two hyphens were used on purpose.